scripts/tests/py/test_metadata.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from metadata import Metadata
+from metadata import Metadata_Reader
+from os import path
+
+
+class TestMetadataReader(unittest.TestCase):
+
+    def setUp(self):
+
+        # build the reader object
+        self.reader=Metadata_Reader(path.join(path.dirname(__file__),"env_test.yaml"))
+            
+
+    def test_parse_single_run(self):
+
+        expected_runs_metadata = {}
+        path_under_test=path.join(path.dirname(__file__),"simple_run_args.txt")
```

Comment:
nit (maybe for future iterations): indirection of tests via a separate file is somewhat inconvenient: we should have more unit tests and if on every test I need to open the txt file it is hard to manage.

it would be easier to read something like:

```py

actual = parse_metadata("""
# not a real text
# runner arguments:
# test-runner-runs: run1
# test-runner-run/run1: app/all-clusters discriminator KVS storage-path commissioning-method discriminator passcode
more
txt
""")

self.assert_eq(
    actual,
    [
        Metadata(
            app="out/linux-x64-all-clusters-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-all-clusters-app",
             discriminator=1234,
             py_script_path=path_under_test,
             run="run1",
             passcode=20202021
        )
    ]
)
```

Reply:
You can either make your parser also accept txt objects besides file paths (seems trivial) or make some helper method that writes a text to a temporary file and parses that (then you don't need to have special worry about how to include txt files inside test builds)